authors/apps/articles/tests/test_filter_articles.py

```diff
@@ -0,0 +1,162 @@
+                         HTTP_AUTHORIZATION='token {}'.format(token)
+                         )
+        self.client.post(self.article_listcreate_url, self.article_data2,
+                         format='json',
+                         HTTP_AUTHORIZATION='token {}'.format(token)
+                         )
+        all_articles = self.client.get(self.article_listcreate, format='json')
+        response = self.client.get(self.filter_title_url,
+                                   format='json',
+                                   HTTP_AUTHORIZATION='token {}'.format(token))
+        self.assertNotEqual(self.article_data['title'], all_articles.data['results'][0]['title'])
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn(self.article_data['title'],
+                      response.data['results'][0]['title'])
+
+    def test_filter_article_tag(self):
+        """Test that you can filter an article with a specific tag"""
+
+        token = self.register()
+        # Create article
+        self.client.post(self.article_listcreate_url, self.article_data,
+                         format='json',
+                         HTTP_AUTHORIZATION='token {}'.format(token)
+                         )
+        self.client.post(self.article_listcreate_url, self.article_data3,
+                         format='json',
+                         HTTP_AUTHORIZATION='token {}'.format(token)
+                         )
+        all_articles = self.client.get(self.article_listcreate, format='json')
+        response = self.client.get(self.filter_tag_url,
+                                   format='json',
+                                   HTTP_AUTHORIZATION='token {}'.format(token))
+        self.assertNotEqual(self.article_data['tagList'], all_articles.data['results'][0]['tagList'])
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertListEqual(self.article_data['tagList'],
+                             response.data['results'][0]['tagList'])
+
+    def test_filter_author(self):
+        """Test that you can filter an article with a specific tag"""
+
+        token = self.register()
+        # Create article
+        self.client.post(self.article_listcreate_url, self.article_data,
+                         format='json',
+                         HTTP_AUTHORIZATION='token {}'.format(token)
+                         )
+        self.client.post(self.article_listcreate_url, self.article_data3,
+                         format='json',
+                         HTTP_AUTHORIZATION='token {}'.format(token)
+                         )
+        all_articles = self.client.get(self.article_listcreate, format='json')
+        response = self.client.get(self.filter_author_url,
+                                   format='json',
+                                   HTTP_AUTHORIZATION='token {}'.format(token))
+        self.assertEqual(self.author_filter, all_articles.data['results'][0]['author'])
```

Comment:
Should these be equal?... all_articles is not a filtered query. I expect assertions to be similar to the others

Reply:
Okay, gotten rod of it.